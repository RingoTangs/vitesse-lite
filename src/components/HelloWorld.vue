<script setup lang="ts">
import { ref } from 'vue'

const props = defineProps<{ initial: number }>()

const n = ref(props.initial)
</script>

<template>
  <div>
    <h1>
      <span>Hello World</span>
      <span> {{ n }}</span>
    </h1>
    <button b="~ 1px solid blue" rounded p-1 type="button" @click="() => n++">
      Increment
    </button>
  </div>
</template>
