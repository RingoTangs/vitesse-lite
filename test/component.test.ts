import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import HelloWorld from '@/components/HelloWorld.vue'

describe('component of HelloWorld.vue', () => {
  it('should render', () => {
    const wrapper = mount(HelloWorld, { props: { initial: 10 } })
    expect(wrapper.text()).toContain('10')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be interactive', async () => {
    const wrapper = mount(HelloWorld, { props: { initial: 0 } })
    expect(wrapper.text()).toContain('0')

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('1')
  })
})
