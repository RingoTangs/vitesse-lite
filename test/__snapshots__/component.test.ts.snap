// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component of HelloWorld.vue > should render 1`] = `
"<div>
  <h1><span>Hello World</span><span>10</span></h1><button type=\\"button\\" b=\\"~ 1px solid blue\\"> Increment </button>
</div>"
`;

exports[`component of TheCounter.vue > should render 1`] = `"<div>10 <button class=\\"inc\\"> + </button><button class=\\"dec\\"> - </button></div>"`;
